test/functional/helpers/createAlloyProxy.js

```diff
@@ -0,0 +1,50 @@
+import { ClientFunction } from "testcafe";
+
+const proxyFunction = ClientFunction((instanceName, command, options) => {
+  return window[instanceName](command, options);
+});
+
+const asyncProxyFunction = ClientFunction((instanceName, command, options) => {
+  window.lastAlloyProxyPromise = window[instanceName](command, options);
+});
+
+const errorMessageProxyFunction = ClientFunction(
+  (instanceName, command, options) => {
+    return window[instanceName](command, options).then(
+      () => undefined,
+      e => e.message
+    );
+  }
+);
+
+const getLastPromise = ClientFunction(() => {
+  return window.lastAlloyProxyPromise;
+});
+
+const getLastErrorMessage = ClientFunction(() => {
+  return window.lastAlloyProxyPromise.then(() => undefined, e => e.message);
+});
+
+const commands = [
+  "configure",
+  "sendEvent",
+  "setConsent",
+  "getIdentity",
+  "setDebug",
+  "getLibraryInfo"
+];
+
+export default (instanceName = "alloy") => {
+  const proxy = {};
+  commands.forEach(command => {
+    proxy[command] = options => proxyFunction(instanceName, command, options);
+    proxy[`${command}Async`] = async options => {
+      await asyncProxyFunction(instanceName, command, options);
+      return { promise: getLastPromise(), errorMessage: getLastErrorMessage() };
```

Comment:
getLastPromise, and getLastErrorMessage are both client functions that return a single promise. The { promise: ..., errorMessage: ...} object is constructed in the node runtime using the serialized promises from testCafe. You are right in that if we were to construct this object inside the ClientFunction, then it would not work.

I will add so comments to this file.

Reply:
Ah, right, I see what you're doing now. Thanks for explaining.